app/hooks/contracts.ts

```diff
@@ -0,0 +1,81 @@
+import * as PolygonCostControllerJson from "core-evm-contracts/deployments/polygon/ActionCostController.json";
+import * as PolygonBountyQuestionJson from "core-evm-contracts/deployments/polygon/BountyQuestion.json";
+import * as PolygonQuestionAPIJson from "core-evm-contracts/deployments/polygon/QuestionAPI.json";
+import * as PolygonQuestionStateControllerJson from "core-evm-contracts/deployments/polygon/QuestionStateController.json";
+import * as PolygonVaultJson from "core-evm-contracts/deployments/polygon/Vault.json";
+import * as PolygonxMetricJson from "core-evm-contracts/deployments/polygon/Xmetric.json";
+import invariant from "invariant";
+import { useMemo } from "react";
+import type { Chain } from "wagmi";
+import { useContract, useNetwork, useProvider } from "wagmi";
+
+export function useContracts({ chainId }: { chainId?: Chain["id"] }) {
+  const { xMetricJson, questionAPIJson, questionStateController, bountyQuestionJson, costController, vaultJson } =
+    useMemo(() => {
+      return getContracts(chainId);
+    }, [chainId]);
+
+  return {
+    xMetricJson,
+    questionAPIJson,
+    questionStateController,
+    bountyQuestionJson,
+    costController,
+    vaultJson,
+  };
+}
+
+function getContracts(chainId?: Chain["id"]) {
```

Comment:
As of the current implementation this doesn't support working locally and using other chains - it's not backwards compatible -  require was used because you can't conditionally import these files (I believe there is a way but I think it makes more sense to use backend) - I'm out a lot of next week but feel like we should try to set up a time to discuss with the group how we want to handle this because it's a big part of the app architecture - the custom hooks in questions.ts I think should work - I would really prefer to avoid the pattern in this file between using imports and loading all the contents on the front end.
Here's a gist of what I envision - https://gist.github.com/jdaly13/a53ebed99926dfd5bb4bfe0aaeb77a16
Basically we get the contracts on the route level and filter them out on the backend so the front end only gets what it needs the abi and the address and any child component can just implement useContext
But I'm certainly open to other alternatives.